lib/graphiti/scope.rb

```diff
@@ -167,5 +176,25 @@ def add_scoping(key, scoping_class, opts, default = {})
       @object = scoping_class.new(@resource, @query.hash, @object, opts).apply
       @unpaginated_object = @object unless key == :paginate
     end
+
+    def deduplicate_entities(resolved)
+      resolved.map! do |entity|
+        next entity unless entity.respond_to?(:id) # Leave no-id (although unusual) entities as is
+
+        if @populated_entities.key?(entity.class)
+          populated_entity = @populated_entities[entity.class].find { |populated_entity| populated_entity&.id == entity.id }
```

Comment:
Using a hash here to lookup the entities would provide O(1) performance vs the current O(n) performance

Reply:
Yeap, will patch soon :)

Reply:
Patched in [8a065ee](https://github.com/graphiti-api/graphiti/pull/476/commits/8a065ee20188d679410d6e8b45d7ac25cf053430)